### PR TITLE
current update is saved when new log is created

### PR DESCRIPTION
**Pull request by ryansrofe**
### Description

When a user creates a log without saving and then tries to create a new log, the first log will be cleared. This update fixes that issue by bringing back the stored values in updates. If a user cancels their update the changes will not be committed.

Fixes # (issue)

#### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

Tested locally in Figma.

### CODE REVIEW

**Review by cpresler — COMMENTED** [review]
Mostly looks good, but there are a couple of places we need to be checking/storing error states so we don't create a validation loop hole.

File: widget-src/components/log/LogDisplay.tsx [hunk 1]
```diff
@@ -92,15 +92,38 @@ export const ChangeLogDisplay = ({
...
+                    links: changeLog.state?.updates?.links ? changeLog.state?.updates?.links : changeLog.links,
+                    link: {
+                      label: changeLog.state?.updates?.link?.label ? changeLog.state?.updates?.link?.label : '',
+                      url: changeLog.state?.updates?.link?.url ? changeLog.state?.updates?.link?.url : '',
+                      icon: changeLog.state?.updates?.link?.icon ? changeLog.state?.updates?.link?.icon : '',
+                      key: changeLog.state?.updates?.link?.key ? changeLog.state?.updates?.link?.key : '',
+                    },
+                    type: changeLog.state?.updates?.type ? changeLog.state?.updates?.type : changeLog.type,
+                    change: changeLog.state?.updates?.change ? changeLog.state?.updates?.change : changeLog.change,
                     linkFormError: { label: false, url: false },
```
[the remark this answers is not in the document]
Comment (ryansrofe) [reply widget-src/components/log/LogDisplay.tsx]: input fields have been refactored to include optional error messaging. The link fields are fully being tracked for errors etc for users and are kept in place if they navigate to another log and come back.

File: widget-src/components/log/LogDisplay.tsx [hunk 2]
```diff
@@ -92,15 +92,38 @@ export const ChangeLogDisplay = ({
...
-                      date: { val: displayDate(changeLog.createdDate, 'date'), er: false },
-                      time: { val: displayDate(changeLog.createdDate, 'time'), er: false },
+                      date: {
+                        val: displayDate(
+                          changeLog.state?.updates?.createdDate
+                            ? changeLog.state?.updates?.createdDate
+                            : changeLog.createdDate,
+                          'date',
+                        ),
+                        er: false,
```
[the remark this answers is not in the document]
Comment (ryansrofe) [reply widget-src/components/log/LogDisplay.tsx]: input fields have been refactored to include optional error messaging. The date and time fields are fully being tracked for errors etc for users and are kept in place if they navigate to another log and come back.

**Comment by ryansrofe** [comment]
Quite a few updates to finish out this PR. Four sets of changes:
- Adding error tracking for Date, Time, Link Label, and URL fields
- New Input Field component to handle all log-based inputs with optional error messaging
- Added focus indicator (darker avatar line) for the log currently being edited
- Fit and finish, adjusting styles and tokens cleanup, removing older unused assets

**Review by cpresler — APPROVED** [review]
Changes look good!

**Pull request merged**